sites/public/pages/listings.tsx

```diff
@@ -1,46 +1,27 @@
 import Head from "next/head"
-import axios from "axios"
-import moment from "moment"
 import {
-  ListingsGroup,
   ListingsList,
   PageHeader,
-  openDateState,
+  AgPagination,
+  AG_PER_PAGE_OPTIONS,
   t,
 } from "@bloom-housing/ui-components"
-import { Listing } from "@bloom-housing/backend-core/types"
 import Layout from "../layouts/application"
 import { MetaTags } from "../src/MetaTags"
-
-export interface ListingsProps {
-  openListings: Listing[]
-  closedListings: Listing[]
-}
-
-const openListings = (listings) => {
-  return listings.length > 0 ? (
-    <ListingsList listings={listings} />
-  ) : (
-    <div className="notice-block">
-      <h3 className="m-auto text-gray-800">{t("listings.noOpenListings")}</h3>
-    </div>
-  )
-}
-
-const closedListings = (listings) => {
-  return (
-    listings.length > 0 && (
-      <ListingsGroup
-        listings={listings}
-        header={t("listings.closedListings")}
-        hideButtonText={t("listings.hideClosedListings")}
-        showButtonText={t("listings.showClosedListings")}
-      />
-    )
-  )
-}
-
-export default function ListingsPage(props: ListingsProps) {
+import { useEffect, useState } from "react"
+import { useListingsData } from "../lib/hooks"
+
+const ListingsPage = () => {
+  /* Pagination state */
+  const [currentPage, setCurrentPage] = useState<number>(1)
+  const [itemsPerPage, setItemsPerPage] = useState<number>(AG_PER_PAGE_OPTIONS[0])
+  const { listingsData, listingsLoading } = useListingsData(currentPage, itemsPerPage)
```

Comment:
How do you plan to overcome the back/forward navigation issue? Can we make these change the URL and have the current page / items per page based on query params?

Reply:
Updated to have the page url change as you navigate forward / back. You can also edit the url directly.